pipeline/plot/behavior_plot.py

```diff
@@ -346,3 +348,99 @@ def compute_insta_phase_amp(data, fs, freq_band=(5, 15)):
         return insta_amp.reshape((trial_count, time_count)), insta_phase.reshape((trial_count, time_count))
     else:
         return insta_amp, insta_phase
+
+
+def get_event_locked_tracking_insta_phase(trials, event, d_name):
+    """
+    Get instantaneous phase of the jaw movement, at the time of the specified "event", for each of the specified "trials"
+    :param trials: query of the SessionTrial - note: the subsequent fetch() will be order_by='trial'
+    :param event: "event" can be
+     + a list of time equal length to the trials - specifying the time of each trial to extract the insta-phase
+     + a single string, representing the event-name to extract the time for each trial
+        In such case, the "event" can be the events in
+            + experiment.TrialEvent
+            + experiment.ActionEvent
+        In the case that multiple of such "event_name" are found in a trial, the 1st one will be selected
+            (e.g. multiple "lick left", then the 1st "lick left" is selected)
+    :param d_name: any attribute name under the tracking.Tracking table - e.g. 'jaw_y', 'tongue_x', etc.
+    :return: list of instantaneous phase, in the same order of specified "trials"
+    """
+
+    trials = trials.proj()
+
+    tracking_fs = (tracking.TrackingDevice & tracking.Tracking & trials).fetch('sampling_rate')
+    if len(set(tracking_fs)) > 1:
+        raise Exception('Multiple tracking Fs found!')
+    else:
+        tracking_fs = float(tracking_fs[0])
+
+    # ---- process the "d_name" input ----
+    nose_types = [n for n in tracking.Tracking.NoseTracking.heading.names if n not in tracking.Tracking.heading.names]
+    tongue_types = [n for n in tracking.Tracking.TongueTracking.heading.names if n not in tracking.Tracking.heading.names]
+    jaw_types = [n for n in tracking.Tracking.JawTracking.heading.names if n not in tracking.Tracking.heading.names]
+
+    if d_name not in nose_types + tongue_types + jaw_types:
+        print(f'Unknown tracking type: {d_name}\nAvailable tracking types are: {nose_types + tongue_types + jaw_types}')
+        return
+
+    for trk_types, trk_tbl in zip((nose_types, tongue_types, jaw_types),
+                                  (tracking.Tracking.NoseTracking, tracking.Tracking.TongueTracking, tracking.Tracking.JawTracking)):
+        if d_name in trk_types:
+            d_tbl = trk_tbl
```

Comment:
these tables are small, I'd just do:
`(tracking.Tracking.NoseTracking * tracking.Tracking.TongueTracking * tracking.Tracking.JawTracking).fetch('foo')` etc, where this is needed later and zap this block